ui/tests/acceptance/oidc-auth-method-test.js

```diff
@@ -2,23 +2,27 @@
  * Copyright (c) HashiCorp, Inc.
  * SPDX-License-Identifier: BUSL-1.1
  */
-
 import { module, test } from 'qunit';
 import { setupApplicationTest } from 'ember-qunit';
-import { click, fillIn, find, waitUntil } from '@ember/test-helpers';
+import { click, fillIn, find, visit, waitUntil } from '@ember/test-helpers';
 import authPage from 'vault/tests/pages/auth';
 import { setupMirage } from 'ember-cli-mirage/test-support';
-import { WindowStub, buildMessage } from 'vault/tests/helpers/oidc-window-stub';
+import { buildMessage, callbackData } from 'vault/tests/helpers/oidc-window-stub';
 import sinon from 'sinon';
 import { Response } from 'miragejs';
 import { setupTotpMfaResponse } from 'vault/tests/helpers/mfa/mfa-helpers';
+import { GENERAL } from '../helpers/general-selectors';
+import { ERROR_MISSING_PARAMS, ERROR_WINDOW_CLOSED } from 'vault/components/auth-jwt';
 
 module('Acceptance | oidc auth method', function (hooks) {
   setupApplicationTest(hooks);
   setupMirage(hooks);
 
   hooks.beforeEach(function () {
-    this.openStub = sinon.stub(window, 'open').callsFake(() => new WindowStub());
+    this.openStub = sinon.stub(window, 'open');
+    // this automatically closes the popup window
+    // some tests return close() as a function because internal logic calls window.close
+    this.openStub.returns({ close: true });
```

Comment:
It seems like the previous `WindowStub` worked in all cases whereas here you need to return either this close object or the function. Curious what the motivation was to update it?

Reply:
It didn't actually work in cases where we needed `close: true`. After a discussion with @emoncuso it seemed sensible to stub a plain javascript object so that it was clearer what was actually necessary for the test.

For example, the previous use of `fakeWindow` was challenging to refactor because it wasn't clear what the test actually required so there was a lot of wasted time spent on trial/erroring. 🫠